doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -102,91 +319,92 @@ Submit pull request
-      ``test/multilocale/``, and ``test/distributions/``.
-
-    * Changes that are likely to cause portability issues should be tested
-      against different platforms and compilers to avoid fallout in the nightly
-      testing to the extent possible.
+* Iterate with the reviewer until you're both satisfied. If you can't come to
+  agreement, one of you should bring other developers (individually or via
+  chapel-developers_) into the conversation to get a broader opinion (one of
+  the jobs of the reviewer is to serve as a proxy for other developers, or to
+  bring those developers into the conversation if they feel unqualified to do
+  so).
 
-  * Note that the quickest way to do testing is to use the parallel testing
-    system across a large number of workstations.  If you have limited testing
-    resources available to you, you can request that a member of the core
-    Chapel team help.
 
-.. _Test System: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst
+.. _Test changes:
 
-.. _Mail chapel-developers@ for review and merge:
+Test changes
+~~~~~~~~~~~~
 
-Mail chapel-developers@ for review and merge
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you did not have the resources to perform at least a full correctness test
+run, this is the point at which a reviewer would do so for you.  See
+`Testing guidance`_ above for more details on what an appropriate amount of
+testing is.
 
-* When you believe your pull request is ready for review, send it to a member
-  of the core Chapel team (or chapel-developers_ if there isn't an obvious
-  person to review it). Even the developers that have write access to the
-  Chapel repository need to have all non-trivial changes reviewed. Developers
-  who have been given write access can merge trivial changes (e.g. small bug
-  fixes, documentation changes) without review.
 
-  .. note::
 
-    Ideally, someone should volunteer to review your pull request within a day
-    or two. If this doesn't happen, feel free to make some noise. Ideally the
-    review should take place within a few days, though timing may vary
-    depending on other deadlines.
+.. _Merge changes in to master:
 
-* Iterate with the reviewer until you're both satisfied. If you can't come to
-  agreement, one of you should bring other developers (individually or via
-  chapel-developers_) into the conversation to get a broader opinion (one of
-  the jobs of the reviewer is to serve as a proxy for other developers, or to
-  bring those developers into the conversation if they feel unqualified to do
-  so).
+Merge changes in to master
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 * Once the pull request is approved, it can be merged. This can be done by
   either the reviewer or developer (given sufficient permissions), as decided
   between the two of them.
 
-  .. note::
+  Note: Recall that while git history can be rewritten, it is difficult and does
```

Comment:
I'd probably just say "It's not generally possible to completely remove a commit from git by the time it makes it in to the master branch. So be very careful...".

I think this should be in the "Getting Ready to Merge" section